### PR TITLE
feat(pkg/trafficpolicy): add routes to outbound traffic policy

### PR DESCRIPTION
**Pull request by michelleN**
**Description**:
feat(pkg/trafficpolicy): add routes to outbound traffic policy

* adds AddRoute method + test to outbound traffic policy
* part of #2034
Please mark with X for applicable areas.

-->
**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

### CODE REVIEW

File: pkg/trafficpolicy/trafficpolicy_test.go [hunk 1]
```diff
@@ -106,10 +116,138 @@ func TestAddRule(t *testing.T) {
...
+					HTTPRouteMatch:   testHTTPRouteMatch,
+					WeightedClusters: set.NewSet(testWeightedCluster),
+				},
+			},
+			expectedErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run("", func(t *testing.T) {
```
Comment (nojnhuh) [inline pkg/trafficpolicy/trafficpolicy_test.go:226]:
```suggestion
		t.Run(tc.name, func(t *testing.T) {
```

**Review by shalier — DISMISSED** [review]
lgtm

**Review by nojnhuh — APPROVED** [review]
one test nit, lgtm

**Review by shashankram — APPROVED** [review]

**Review by shashankram — APPROVED** [review]

File: pkg/trafficpolicy/trafficpolicy.go [hunk 2]
```diff
@@ -27,3 +28,31 @@ func (in *InboundTrafficPolicy) AddRule(route RouteWeightedClusters, allowedServ
...
+//	already exists, an error will be returned. If a Route with the given HTTP route match does not exist,
+//	a Route with the given HTTP route match and weighted clusters will be added to the Routes on the OutboundTrafficPolicy
+func (out *OutboundTrafficPolicy) AddRoute(httpRouteMatch HTTPRouteMatch, weightedClusters ...service.WeightedCluster) error {
+	routeExists := false
+	wc := set.NewSet()
+	for _, c := range weightedClusters {
+		wc.Add(c)
+	}
+
+	for _, existingRoute := range out.Routes {
```
Comment (ksubrmnn) [inline pkg/trafficpolicy/trafficpolicy.go:42]: What scenario is this for? Would it be possible that the weightedclusters passed in would contain new wc's that need to be added? 

Can you add a comment with the scenario?

**Review by ksubrmnn — CHANGES_REQUESTED** [review]

**Review by ksubrmnn — APPROVED** [review]

**Review by shalier — APPROVED** [review]

**Pull request merged**